### DISCUSSION TITLE
QST: simplify resulting gaps between adjacent polygons

### DISCUSSION THREAD

**Discussion by KarenChen9999** [geopandas/geopandas#2721] [Q&A]
I used simplify from geopanda to smooth polygons by preserving topology. When I used a low tolerance i.e. 0.0001, it preserved topology. However when i tried a slightly high tolerance i.e. 0.001, it left gaps between adjacent polygons. How can i preserve topology with any tolerance number?

```python
# no issue with topology with tolerance of 0.0001
shape1=gpd.read_file(_shape1)
shape1=shape1.to_crs({'init': 'epsg:4326'})
shape1["geo"] = shape1.simplify(tolerance=0.0001, preserve_topology=True)

# issue with topology with tolerance of 0.001
shape1=gpd.read_file(_shape1)
shape1=shape1.to_crs({'init': 'epsg:4326'})
shape1["geo"] = shape1.simplify(tolerance=0.001, preserve_topology=True)

```

**Comment by martinfleis** [comment]
Hi, that is not what `preserve_topology` means. It preserves topology of individual geometries. You can use `topojson` to simplify an array while preserving topological relations between polygons. See #1387.

**Comment by KarenChen9999** [comment]
Hi, thanks for the prompt comments. it would be good to add this function to geopandas. And indeed the preserve_topology is a bit confused.
